velox/functions/sparksql/String.h

```diff
@@ -583,4 +583,167 @@ struct OverlayVarbinaryFunction : public OverlayFunctionBase {
   }
 };
 
+/// translate(string, match, replace) -> varchar
+///
+///   Returns a new translated string. It translates the character in ``string``
+///   by a character in ``replace``. The character in ``replace`` is
+///   corresponding to the character in ``match``. The translation will
+///   happen when any character in ``string`` matching with a character in
+///   ``match``.
+template <typename T>
+struct TranslateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // ASCII input always produces ASCII result.
+  static constexpr bool is_default_ascii_behavior = true;
+
+  folly::F14FastMap<std::string, std::string> dictionaryUtf8_;
+  folly::F14FastMap<char, char> dictionaryAscii_;
```

Comment:
nit: consider renaming to asciiDictionary_ and unicodeDictionary_. This way the reader gets to the important information faster (ascii vs. unicode) .